pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -65,25 +65,21 @@ func ApplyContexts(spec *v1beta1.PipelineSpec, pipelineName string, pr *v1beta1.
 	return ApplyReplacements(spec, replacements, map[string][]string{})
 }
 
-// ApplyTaskResults applies the ResolvedResultRef to each PipelineTask.Params in targets
+// ApplyTaskResults applies the ResolvedResultRef to each PipelineTask.Params and Pipeline.WhenExpressions in targets
 func ApplyTaskResults(targets PipelineRunState, resolvedResultRefs ResolvedResultRefs) {
-	stringReplacements := map[string]string{}
-
-	for _, resolvedResultRef := range resolvedResultRefs {
-		replaceTarget := fmt.Sprintf("%s.%s.%s.%s", v1beta1.ResultTaskPart, resolvedResultRef.ResultReference.PipelineTask, v1beta1.ResultResultPart, resolvedResultRef.ResultReference.Result)
-		stringReplacements[replaceTarget] = resolvedResultRef.Value.StringVal
-	}
+	replacements := resolvedResultRefs.getReplacements()
```

Comment:
nice!

(if you wanted, this improvement would probably be a good candidate for a separate PR as well? but np if you dont want to bother esp. since this PR is getting so close to being merged)

Reply:
moved this refactor to a separate PR: https://github.com/tektoncd/pipeline/pull/3169